projectforum/static/js/project-detail.js

```diff
@@ -1,3 +1,44 @@
+function addBookmark () {
+    var projectid;
+    projectid = $(this).attr("data-projectid");
+    $.get('/project/'+projectid+'/bookmark_add/', function(data){
+        $('a#bookmark')
+            .addClass('active')
+            .attr('title','Remove from bookmarks')
+            .unbind('click')
+            .bind('click', removeBookmark);
+            });
+
+}
+
+function removeBookmark () {
+    var projectid;
+    projectid = $(this).attr("data-projectid");
+    $.get('/project/'+projectid+'/bookmark_remove/', function(data){
+        $('a#bookmark')
+            .removeClass('active')
+            .attr('title','Bookmark this project')
+            .unbind('click')
+            .bind('click', addBookmark);
+        });
+}
+
+function setBookmark(bookmarked){
+    if (bookmarked) {
+        $('a#bookmark')
+            .addClass('active')
+            .attr('title','Remove from bookmarks')
+            .unbind('click')
+            .bind('click', removeBookmark);
+    } else {
+        $('a#bookmark')
+            .removeClass('active')
+            .attr('title','Bookmark this project')
+            .unbind('click')
+            .bind('click', addBookmark);
+    }
+}
```

Comment:
This has a lot of duplicated code. Instead of relying on a function in the script tag above, can we move this to the document ready section below and just apply a click callback to the bookmark there? The link can have the class active depending on whether it is bookmark and the click callback can act based off of whether the class is present.